crates/matrix-sdk-common/src/linked_chunk/relational.rs

```diff
@@ -345,13 +347,36 @@ where
         };
 
         // Find the last chunk.
-        let Some(chunk_row) = self
-            .chunks
-            .iter()
-            .find(|chunk_row| chunk_row.room_id == room_id && chunk_row.next_chunk.is_none())
-        else {
-            // Chunk is not found.
-            return Ok((None, chunk_identifier_generator));
+        let chunk_row = match self.chunks.iter().try_fold(0, |mut number_of_chunks, chunk_row| {
```

Comment:
Can you use a plain loop instead of `try_fold`? I don't understand what this code does (what does it mean to continue with a value?), so I'd rather have a plain loop, so as to not confuse readers.